### DISCUSSION TITLE
feat: ml model scanning

### DISCUSSION THREAD

**Discussion by terekete** [aquasecurity/trivy#5185] [Ideas]
### Description

- appears there could be a vector of attack where someone might embed code in ML model, is there a way to scan models using trivy?

### Target

Container Image

### Scanner

Vulnerability

**Comment by nikpivkin** [comment]
Hi @terekete !

Trivy doesn't support scanning ML models. But I'm wondering if this kind of attack has already happened and if there is any research in this area?

**Reply by terekete** [reply]
Seems it could be an attack vector because most models are provided as pickle files.
Pickles can embed code. Alt. there are some mentions here on the subject: https://huggingface.co/docs/hub/security-pickle
